api_subscription_management.go

```diff
@@ -2,26 +2,148 @@ package deribit
 
 import (
 	"context"
+	"strings"
 
 	"github.com/KyberNetwork/deribit-api/models"
 )
 
-func (c *Client) PublicSubscribe(ctx context.Context, params *models.SubscribeParams) (result models.SubscribeResponse, err error) {
+func (c *Client) Subscribe(channels []string) error {
+	return c.subscribe(channels, true)
+}
+
+func (c *Client) subscribe(channels []string, isNewSubscription bool) error {
+	l := c.l.With("func", "subscribe")
+	var publicChannels []string
+	var privateChannels []string
+
+	c.mu.RLock()
+	currentMap := c.subscriptionsMap
```

Comment:
This can have some race conditions when call in multiple go routines. I think we need to read and update the subscription mapping in one critical section.